includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1342,17 +1368,34 @@ public function get_contact_data( $email, $return_details = false ) {
 				'status'     => $contact['status'],
 			];
 		}
+		$this->contacts_data[ $email ] = $data;
 		return $data;
 	}
 
+	/**
+	 * A memoized version of the get_contact_data method.
+	 *
+	 * To be used in methods where it's not critical that we have the most updated version of a contact
+	 *
+	 * @param string $email The contact email.
+	 * @return array|WP_Error Response or error if contact was not found.
+	 */
+	public function get_cached_contact_data( $email ) {
+		if ( isset( $this->contacts_data[ $email ] ) && ! is_wp_error( $this->contacts_data[ $email ] ) ) {
```

Comment:
Feels a bit strange to allow errors to exist in contact data cache. Should we be handling these errors before they are populated here, and maybe filtering these entries out?

Reply:
> Should we be handling these errors before they are populated here, and maybe filtering these entries out?

We actually are, if you look at the method above. I added this check before realizing it. I'll remove this check :+1:

Reply:
removed in https://github.com/Automattic/newspack-newsletters/pull/1433/commits/61819ec55dc9c1e22c1c25d77e8cf5558f043f13